### PR TITLE
Stream imagebuild logs directly from pods

### LINKED ISSUE #7874: make imagebuild logs independent of cluster locality
Context: #7435

Currently imagebuilds are delivered via `server` -> `image-builder-mk3`. This was the backwards-compatible way from `mk2`, but requires `server` and `image-builder_mk3` to reside in the same cluster (= region).

With this issue the plan is to re-use the per-workspace log-streaming mechanism we already have in place for prebuilds. This enables:
 - removing the last bits of cross-cluster messagebus
 - the move of image-builder into the workspace cluster

### PR DESCRIPTION
**Pull request by geropl**
## Description
Streams log URLs from the `imagebuild` workspace directly instead via `image-builder-mk3`, which made it cluster dependent.

Basically
 - old: `server` -> (cluster-local) `image-builder-mk3` -> `imgbuild` workspace
 - new: `server` -> `imgbuild` workspace

## Related Issue(s)

Fixes #7874

## How to test
- start an image build: https://gpl-7874-imagebuild-logs.staging.gitpod-dev.com/#https://github.com/geropl/roborally
- notice how the image build starts and logs are streamed shortly after. Once the build finished (failed), the state is reflected accordingly
- note: connecting to an already running image build does not seem to work atm because the `build(...)` call never returns from imagebuilder. I tried to find the root cause, but ultimately it's out-of-scope here.

## Release Notes

```release-note
none
```

## Documentation

- [ ] /werft with-clean-slate-deployment
- [x] /werft with-observability

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/image-builder-api/imgbuilder.proto
Comment (csweichel) [inline components/image-builder-api/imgbuilder.proto:132]: A matter of taste, but I'd wrap this in a message of its own, e.g.
```protobuf
    LogInfo logs = 6;
}

message LogInfo {
    string url = 1;
    map<string, string> header = 7;
}
```

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (csweichel) [inline components/server/src/workspace/gitpod-server-impl.ts:1131]: What's the behaviour of those things don't happen within 20s?
Can we get away without the time-based behaviour?

File: components/image-builder-api/imgbuilder.proto
Comment (geropl) [reply components/image-builder-api/imgbuilder.proto:132]: I'd prefer that myself - also to mirror the code that's already there. :+1:

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:1131]: > What's the behaviour of those things don't happen within 20s?

The (browser) client re-trys (if this is still wanted).

> Can we get away without the time-based behaviour?

Only without db-sync, and only if we had a way to know the definite state of an `imagebuild`. We either:
 - enhance the field `workspace.imageBuildInfo` to also contain info about the state of the build (pending, pulling, running, stopped)
 - or we re-visit our decision to surface `imagebuild` on the application side (e..g, DB), so we could read that state easily

Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:1131]: @csweichel :point_up: This would also the lifecycle-mismatch of `imageBuildInfo`: The current change stores it in `Workspace` because it is convenient. And this is correct as long as we never have to re-build an image. But once we have to - think of docker image cleanup, registry data loss, SH user re-sets a PV, etc. - we're updating the workspace again, which is kind of weird.

Having a own entity for this akin to prebuilds would really make sense. Conceptually this could also help with explaining image builds better to users by providing more transparency in the UI.

I'm not 100% settled we want that (and whether/when the effort is justified), just this PR got me thinking in that direction again.

Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:1131]: Another argument: The regionality split between imagebuilds and workspaces. With this PR we potentially override `imageBuildInfo` from different regions when building images with different `image-builder-mk3` instances.

Comment (csweichel) [reply components/server/src/workspace/gitpod-server-impl.ts:1131]: what if we stored the `imageBuildInfo` on the instance instead of the workspace? That's much cluster to the actual lifecycle of the build, as every instance can potentially require an image build.

Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:1131]: Yes, that would fix problem 2 and 3 (the lifecycle and regionality "issues"). :+1: Still we'll have the "not enough state" information. I will go with that approach for this PR.

We can think about how to surface image build later. But IMO this is sth we should solve, also to help make image builds more transparent to the user. For instance by offering more UI that explains:
 - when an image build happened
 - what triggered it
 - why it failed
 - how to fix it

BTW this is connected to "prebuild reliability" /cc @jldec

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e713670d411534f3e37f15ad31ed52b2b99304c1</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-869073898" title="Approved">JanKoehnlein</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-869162528" title="Approved">JanKoehnlein</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
did not actually try this, but image-builder and API changes look good.

/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 5ae979b6ca3262a6e075e23062b58251338adcde</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7899#issuecomment-1028087456" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-869162528" title="Approved">JanKoehnlein</a>*

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- ~~[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)~~ [csweichel]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by sagor999 — APPROVED** [review]

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7899#issuecomment-1028087456" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-869162528" title="Approved">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-870768210" title="Approved">sagor999</a>*

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- ~~[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)~~ [csweichel,sagor999]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7899#issuecomment-1028087456" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-870817131" title="Approved">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7899#pullrequestreview-870768210" title="Approved">sagor999</a>*

Associated issue: *#7874*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [JanKoehnlein]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- ~~[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)~~ [csweichel,sagor999]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
